pydantic/_internal/_generate_schema.py

```diff
@@ -1493,6 +1506,7 @@ def _get_wrapped_inner_schema(
         def new_handler(source: Any) -> core_schema.CoreSchema:
             schema = metadata_get_schema(source, get_inner_schema)
             schema = self.apply_single_annotation(schema, annotation)
+            schema = self.apply_single_annotation_json_schema(schema, annotation)
```

Comment:
As you can see here, everything related to JSON schema now runs _after_ everything not related to JSON schema.

Reply:
Typo?

Reply:
No typo as far as I can tell — the `self.apply_single_annotation` does everything not related to JSON schema, then after that, `self.apply_single_annotation_json_schema` does everything related to json schema. I think that's what I said in the comment above?